openid-4-verifiable-credential-issuance-1_0.md

```diff
@@ -804,6 +827,9 @@ This specification defines the following proof types:
 * `jwt`: A JWT [@!RFC7519] is used for proof of possession. When a `proof_type` parameter in a `proof` object is set to `jwt`, it MUST also contain a `jwt` parameter that includes a JWT as defined in (#jwt-proof-type). When a `proofs` object is using a `jwt` proof type, it MUST include a `jwt` parameter with its value being an array of JWTs, where each JWT is formed as defined in (#jwt-proof-type).
 * `ldp_vp`: A W3C Verifiable Presentation object signed using the Data Integrity Proof [@VC_Data_Integrity] as defined in [@VC_DATA_2.0] or [@VC_DATA] is used for proof of possession. When a `proof_type` parameter in a `proof` object is set to `ldp_vp`, it MUST also contain an `ldp_vp` parameter that includes a [W3C Verifiable Presentation](https://www.w3.org/TR/vc-data-model-2.0/#presentations-0) defined in (#ldp-vp-proof-type). When a `proofs` object is using a `ldp_vp` proof type, it MUST include an `ldp_vp` parameter with its value being an array of [W3C Verifiable Presentations](https://www.w3.org/TR/vc-data-model-2.0/#presentations-0), where each of these W3C Verifiable Presentation is formed as defined in (#ldp-vp-proof-type).
 
+Additional proof types MAY be defined and used.
+Proof types used MUST be understood by both the Wallet and the Credential Issuer.
```

Comment:
Not that I don't agree with this, I am just curious to see why this text is formulated in this way as this one breaks the pattern of "receiver decides".

I ask because a Credential Issuer can contain a wide set of optional `proof_type`s, and I expected a sentence like "Wallet MUST ignore any unrecognized proof types from Credential Issuer Metadata."
"Credential Issuer MUST reject any Credential Request containing an unrecognized proof type."

Reply:
The normal "MUST ignore any unrecognized parameters" language would be wrong here, because it's a security issue if you can't verify the proof.  But this is nonetheless an extension point, so I wanted to be explicit about the rules for handling this kind of extension.

Reply:
Understood. I just wondered because unilateral extension where a Credential Issuer defines a new proof type the spec and offers more to a wallet that is oblivious to the extension in my mind did not fit "MUST be understood by both"

Reply:
we should discuss this. I agree the MUST here feels different

Reply:
I also agree with the intent of this MUST text. Maybe expand it a bit more to better describe what using a proof type entails?:

```suggestion
Proof types MUST be recognized and supported by both the Wallet and the Credential Issuer in order to bind an issued Credential to specific cryptographic key material.
```